### PR TITLE
feat(common): add injection token for default `DatePipe` configuration

### PR DESCRIPTION
**Pull request by matthiasweiss**
In order to globally change the default format of the Angular date pipe, a new injection token called "DATE_PIPE_DEFAULT_FORMAT" is added and used in the date pipe

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently the date pipe defaults to the `mediumDate` format and there is no way to override this behavior with an injection token. If it is not present, the behavior is identical.
 
## What is the new behavior?
The `DATE_PIPE_DEFAULT_FORMAT` injection token can be used to override the default format that the Angular date pipe uses.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Since there exist no tests for the `DATE_PIPE_DEFAULT_TIMEZONE` injection token, which behaves alomst identically to the `DATE_PIPE_DEFAULT_FORMAT`, there are also no tests for `DATE_PIPE_DEFAULT_FORMAT`.

### CODE REVIEW

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for the PR. It looks like this feature was requested in #39194, but the ticket was closed with the proposal to add a custom pipe based on DatePipe.

However, since we've added the `DATE_PIPE_DEFAULT_TIMEZONE` later in #43611, I think adding a token to configure date format makes sense and makes the DatePipe API more consistent.

We'll discuss it further with the team and I'll post an update.

> Since there exist no tests for the DATE_PIPE_DEFAULT_TIMEZONE injection token, which behaves alomst identically to the DATE_PIPE_DEFAULT_FORMAT, there are also no tests for DATE_PIPE_DEFAULT_FORMAT.

PR #43611 (where the `DATE_PIPE_DEFAULT_TIMEZONE` was implemented) has some relevant tests. Could you please add similar ones? I think it'd be also great to have a test where a component is created via TestBed and uses DatePipe in a template, so that the `DATE_PIPE_DEFAULT_FORMAT` token behavior is tested additionally.

**Comment by matthiasweiss** [comment]
@AndrewKushnir thanks for the note on the existing tests, I just globally searched for the token and there was nothing there so I thought there are none, my bad. I'll try to add tests sometime this week, should not take too long anyway!

File: packages/common/src/pipes/date_pipe.ts [hunk 1]
```diff
@@ -18,6 +18,12 @@ import {invalidPipeArgumentError} from './invalid_pipe_argument_error';
  */
 export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_DEFAULT_TIMEZONE');
 
+/**
+ * Optionally-provided default format to use for all instances of `DatePipe` (such as `short`).
+ * If the value isn't provided, the `DatePipe` will default to 'mediumDate'.
+ */
+export const DATE_PIPE_DEFAULT_FORMAT = new InjectionToken<string>('DATE_PIPE_DEFAULT_FORMAT');
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:25]: @matthiasweiss we've discussed this feature with the Angular team and we'd like to proceed with the change.

The proposal is to introduce a new token that would represent a config for the DatePipe (containing the necessary  fields) vs introducing tokens for each config option. See an example in Material Dialog:
https://github.com/angular/components/blob/b98ae5ac07e421e868389af5f02fa92d2aa9cccf/src/material/dialog/dialog.ts#L34

We can call the token like `DATE_PIPE_DEFAULT_OPTIONS` and it'd represent an object/interface with 2 fields:
- dateFormat
- timezone

as a part of this change (in a separate commit), we can deprecate the `DATE_PIPE_DEFAULT_TIMEZONE` token (while still keep using it in the code for backwards-compatibility).

Please let me know if you have any questions or any additional information is required, I'll be happy to help. Thank you.

Comment (matthiasweiss) [reply packages/common/src/pipes/date_pipe.ts:25]: That's great news! 😄 

Just one quick question regarding the deprecation: is adding a doc block with `@deprecated` and a description sufficient?

Comment (AndrewKushnir) [reply packages/common/src/pipes/date_pipe.ts]: > Just one quick question regarding the deprecation: is adding a doc block with @deprecated and a description sufficient?

There are 3 places where the deprecation should be mentioned:

- the code itself: annotate the symbol with the `@deprecated` tag and mention what to use instead as a description (for ex. `@deprecated use DATE_PIPE_DEFAULT_OPTIONS token to configure DatePipe`)
- the commit message should include a deprecation note with additional info and a recommendation on what should be used instead, see commit messages here as an example: https://github.com/angular/angular/pull/44749/commits
- the Deprecations guide should be updated with similar information: see aio/content/guide/deprecations.md.

Thank you.

File: packages/common/src/pipes/date_pipe.ts [hunk 2]
```diff
@@ -18,6 +19,13 @@ import {invalidPipeArgumentError} from './invalid_pipe_argument_error';
  */
 export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_DEFAULT_TIMEZONE');
 
+/**
+ * Optionally-provided default format to use for all instances of `DatePipe` (such as `short`).
+ * If the value isn't provided, the `DatePipe` will default to 'mediumDate'.
+ */
+export const DATE_PIPE_DEFAULT_OPTIONS =
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:26]: Let's add the public API annotation to this new symbol:
```suggestion
/**
 * Optionally-provided default format to use for all instances of `DatePipe` (such as `short`).
 * If the value isn't provided, the `DatePipe` will default to 'mediumDate'.
 *
 * @publicApi
 */
export const DATE_PIPE_DEFAULT_OPTIONS =
```

File: packages/common/src/pipes/date_pipe_config.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export interface DatePipeConfig {
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe_config.ts:9]: Note: this symbol should also become a part of the public API. We'd need to add docs for the symbol itself (with example usages) and the fields.
```suggestion
/**
 * ... docs ...
 *
 * @publicApi
 */
export interface DatePipeConfig {
```

File: packages/common/src/pipes/index.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@
  */
 import {AsyncPipe} from './async_pipe';
 import {LowerCasePipe, TitleCasePipe, UpperCasePipe} from './case_conversion_pipes';
-import {DATE_PIPE_DEFAULT_TIMEZONE, DatePipe} from './date_pipe';
+import {DATE_PIPE_DEFAULT_OPTIONS, DATE_PIPE_DEFAULT_TIMEZONE, DatePipe} from './date_pipe';
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/index.ts:16]:
```suggestion
import {DATE_PIPE_DEFAULT_OPTIONS, DATE_PIPE_DEFAULT_TIMEZONE, DatePipe} from './date_pipe';
import {DatePipeConfig} from './date_pipe_config';
```

File: packages/common/src/pipes/index.ts [hunk 5]
```diff
@@ -24,6 +24,7 @@ import {SlicePipe} from './slice_pipe';
 export {
   AsyncPipe,
   CurrencyPipe,
+  DATE_PIPE_DEFAULT_OPTIONS,
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/index.ts:27]:
```suggestion
  DATE_PIPE_DEFAULT_OPTIONS,
  DatePipeConfig,
```

File: packages/common/src/pipes/date_pipe.ts [hunk 6]
```diff
@@ -207,13 +218,14 @@ export class DatePipe implements PipeTransform {
...
   transform(
-      value: Date|string|number|null|undefined, format = 'mediumDate', timezone?: string,
+      value: Date|string|number|null|undefined, format?: string, timezone?: string,
       locale?: string): string|null {
     if (value == null || value === '' || value !== value) return null;
 
     try {
       return formatDate(
-          value, format, locale || this.locale, timezone ?? this.defaultTimezone ?? undefined);
+          value, format ?? this.defaultOptions?.dateFormat ?? 'mediumDate', locale || this.locale,
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:227]: It'd be great to move the `'mediumDate'` to a const (and store it in the `date_pipe_config.ts`):
```suggestion
          value, format ?? this.defaultOptions?.dateFormat ?? DEFAULT_DATE_FORMAT, locale || this.locale,
```

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for additional updates, I've left a few more comments. Thank you.

File: packages/common/src/pipes/date_pipe_config.ts (hunk 3 above)
Comment (matthiasweiss) [reply packages/common/src/pipes/date_pipe_config.ts:9]: Thanks for the comments, I tried my best with the usage notes!

Also I skipped the deprecation commit for now, I should be able to do that sometime this week tho 😄

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for updating the PR, this change is very close to a ready state! 👍 

I've left a few comments, please take a look when you get a chance.

Thank you.

**Comment by matthiasweiss** [comment]
@AndrewKushnir I've added the deprecation commit yesterday (thanks for the example commits, those were very useful)

Unfortunately I don't know how to fix the problem with the `ci/circleci:test` job, apparently the build is a little too big 😢

Is this even a problem and if so, do you have any ideas how it could be fixed?

File: packages/common/test/pipes/date_pipe_spec.ts [hunk 7]
```diff
@@ -72,9 +72,31 @@ import {TestBed} from '@angular/core/testing';
...
 
     describe('transform', () => {
-      it('should use "mediumDate" as the default format',
+      it('should use "mediumDate" as the default format if no format is provided',
          () => expect(pipe.transform('2017-01-11T10:14:39+0000')).toEqual('Jan 11, 2017'));
 
+      it('should give precedence to the passed in format',
+         () => expect(pipe.transform('2017-01-11T10:14:39+0000', 'shortDate')).toEqual('1/11/17'));
+
+      it('should use provided format as default format when no format is passed in', () => {
```
[the remark this answers is not in the document]
Comment (matthiasweiss) [reply packages/common/test/pipes/date_pipe_spec.ts]: Thanks for all the comments, added a commit that addresses all of your suggestions!

File: aio/content/guide/deprecations.md [hunk 8]
```diff
@@ -107,6 +108,7 @@ In the [API reference section](api) of this site, deprecated APIs are indicated
 |:---                                                                                           |:---                                                 |:---                   |:---     |
 | [`CurrencyPipe` - `DEFAULT_CURRENCY_CODE`](api/common/CurrencyPipe#currency-code-deprecation) | `{provide: DEFAULT_CURRENCY_CODE, useValue: 'USD'}` | v9                    | From v11 the default code will be extracted from the locale data given by `LOCALE_ID`, rather than `USD`. |
 | [`NgComponentOutlet.ngComponentOutletNgModuleFactory`](api/common/NgComponentOutlet)          | `NgComponentOutlet.ngComponentOutletNgModule`       | v14                   | Use the `ngComponentOutletNgModule` input instead. This input doesn't require resolving NgModule factory. |
+| [`DatePipe` - `DATE_PIPE_DEFAULT_TIMEZONE`](api/common/DATE_PIPE_DEFAULT_TIMEZONE) |`{ provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: { timezone: '-1200' }` | v14                    | Use the `DATE_PIPE_DEFAULT_OPTIONS` injection token, which can configure multiple settings at once, instead. |
```
Comment (AndrewKushnir) [inline aio/content/guide/deprecations.md:111]:
```suggestion
| [`DatePipe` - `DATE_PIPE_DEFAULT_TIMEZONE`](api/common/DATE_PIPE_DEFAULT_TIMEZONE) |`{ provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: { timezone: '-1200' }` | v15                    | Use the `DATE_PIPE_DEFAULT_OPTIONS` injection token, which can configure multiple settings at once instead. |
```

File: aio/content/guide/deprecations.md [hunk 9]
```diff
@@ -85,6 +85,7 @@ v14 - v17
 | `@angular/service-worker`           | [`SwUpdate#available`](api/service-worker/SwUpdate#available)                                              | <!-- v13 --> v16         |
 | template syntax                     | [`/deep/`, `>>>`, and `::ng-deep`](#deep-component-style-selector)                                         | <!--  v7 --> unspecified |
 | template syntax                     | [`bind-`, `on-`, `bindon-`, and `ref-`](#bind-syntax)                                                      | <!-- v13 --> v15         |
+| `@angular/common`                   | [`DatePipe` - `DATE_PIPE_DEFAULT_TIMEZONE`](api/common/CurrencyPipe#currency-code-deprecation)              | <!--  v14 --> v17         |
```
Comment (AndrewKushnir) [inline aio/content/guide/deprecations.md:88]:
```suggestion
| `@angular/common`                   | [`DatePipe` - `DATE_PIPE_DEFAULT_TIMEZONE`](api/common/CurrencyPipe#currency-code-deprecation)              | <!--  v15 --> v17         |
```

File: packages/common/src/pipes/date_pipe.ts [hunk 10]
```diff
@@ -10,14 +10,46 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
+ *
+ * Override the default date format by providing it in a module/component:
+ * ```typescript
+ * {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {dateFormat: 'shortDate'}}
+ * ```
+ *
+ * Override the default timezone by providing it in a module/component:
+ * ```typescript
+ * {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {timezone: '-1200'}}
+ * ```
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:48]:
```suggestion
 * Override the default date format by providing a value using the token:
 * ```typescript
 * providers: [
 *   {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {dateFormat: 'shortDate'}}
 * ]
 * ```
 *
 * Override the default timezone by providing a value using the token:
 * ```typescript
 * providers: [
 *   {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {timezone: '-1200'}}
 * ]
 * ```
```

File: packages/common/src/pipes/date_pipe.ts [hunk 11]
```diff
@@ -10,14 +10,46 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
+ *
+ * @deprecated use DATE_PIPE_DEFAULT_OPTIONS token to configure DatePipe
  */
 export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_DEFAULT_TIMEZONE');
 
+/**
+ * DI token that allows to provide default configuration for the `DatePipe` instances in an
+ * application. The value is an object which can include the following fields:
+ * - `dateFormat`: configures the default date format. If the value is not provided, the `DatePipe`
+ * will use 'DEFAULT_DATE_FORMAT'.
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:28]:
```suggestion
 * - `dateFormat`: configures the default date format. If the not provided, the `DatePipe`
 * will use the 'mediumDate' as a value.
```

File: packages/common/src/pipes/date_pipe.ts [hunk 12]
```diff
@@ -185,19 +217,25 @@ export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_
...
+  ) {}
 
   /**
    * @param value The date expression: a `Date` object,  a number
    * (milliseconds since UTC epoch), or an ISO string (https://www.w3.org/TR/NOTE-datetime).
-   * @param format The date/time components to include, using predefined options or a
-   * custom format string.
+   * @param options The date/time components to include, using predefined options or a
+   * custom format string.  When not supplied, either the `dateFormat` property of the
+   * `DATE_PIPE_DEFAULT_OPTIONS` injection token is used or the 'DEFAULT_DATE_FORMAT'.
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:229]:
```suggestion
   * custom format string.  When not provided, the `DatePipe` looks for the value using the
   * `DATE_PIPE_DEFAULT_OPTIONS` injection token (and reads the `dateFormat` property).
   * If the token is not configured, the 'mediumDate' is used as a value.
```

File: packages/common/src/pipes/date_pipe.ts [hunk 13]
```diff
@@ -185,19 +217,25 @@ export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_
...
    * @param timezone A timezone offset (such as `'+0430'`), or a standard UTC/GMT, or continental US
-   * timezone abbreviation. When not supplied, either the value of the `DATE_PIPE_DEFAULT_TIMEZONE`
-   * injection token is used or the end-user's local system timezone.
+   * timezone abbreviation. When not supplied, either the `timezone` property of the
+   * `DATE_PIPE_DEFAULT_OPTIONS` injection token is used or the end-user's local system timezone.
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:232]: Could you please also rephrase this similar to the `format` field docs above (after applying the changes there)?

File: packages/common/src/pipes/date_pipe.ts [hunk 14]
```diff
@@ -185,19 +217,25 @@ export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_
...
+      @Inject(DATE_PIPE_DEFAULT_TIMEZONE) @Optional() private defaultTimezone?: string|null,
+      @Inject(DATE_PIPE_DEFAULT_OPTIONS) @Optional() private defaultOptions?: DatePipeConfig|null,
+  ) {}
 
   /**
    * @param value The date expression: a `Date` object,  a number
    * (milliseconds since UTC epoch), or an ISO string (https://www.w3.org/TR/NOTE-datetime).
-   * @param format The date/time components to include, using predefined options or a
-   * custom format string.
+   * @param options The date/time components to include, using predefined options or a
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:227]:
```suggestion
   * @param format The date/time components to include, using predefined options or a
```

File: packages/common/src/pipes/date_pipe_config.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Optionally provided date pipe configuration.
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe_config.ts:10]:
```suggestion
 * An interface that describes the date pipe configuration, which can be provided using the `DATE_PIPE_DEFAULT_OPTIONS` token.
```

File: packages/common/src/pipes/date_pipe_config.ts [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+ * Override the default date format by providing it in a module/component:
+ * ```typescript
+ * {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {dateFormat: 'shortDate'}}
+ * ```
+ *
+ * Override the default timezone by providing it in a module/component:
+ * ```typescript
+ * {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {timezone: '-1200'}}
+ * ```
+ *
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe_config.ts:27]: Let's drop the usage notes from here and point to the `DATE_PIPE_DEFAULT_OPTIONS` docs instead (so we don't have duplication of docs):
```suggestion
 * @see `DATE_PIPE_DEFAULT_OPTIONS`
```

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for additional changes. I've left a few more comments, mostly related to docs. Please let us know (by leaving a comment in this thread) once the feedback is addressed and we can do the final code-review round (will need some extra LGTMs, since this change affects the public API). Thank you.

**Comment by matthiasweiss** [comment]
@AndrewKushnir your comments should be reflected in the PR now, thanks again for the feedback, hopefully all issues are addressed now!

File: packages/common/src/common.ts [hunk 17]
```diff
@@ -22,7 +22,7 @@ export {parseCookieValue as ɵparseCookieValue} from './cookie';
 export {CommonModule} from './common_module';
 export {NgClass, NgForOf, NgForOfContext, NgIf, NgIfContext, NgPlural, NgPluralCase, NgStyle, NgSwitch, NgSwitchCase, NgSwitchDefault, NgTemplateOutlet, NgComponentOutlet} from './directives/index';
 export {DOCUMENT} from './dom_tokens';
-export {AsyncPipe, DatePipe, DATE_PIPE_DEFAULT_TIMEZONE, I18nPluralPipe, I18nSelectPipe, JsonPipe, LowerCasePipe, CurrencyPipe, DecimalPipe, PercentPipe, SlicePipe, UpperCasePipe, TitleCasePipe, KeyValuePipe, KeyValue} from './pipes/index';
+export {AsyncPipe, DatePipe, DATE_PIPE_DEFAULT_TIMEZONE, DATE_PIPE_DEFAULT_OPTIONS, I18nPluralPipe, I18nSelectPipe, JsonPipe, LowerCasePipe, CurrencyPipe, DecimalPipe, PercentPipe, SlicePipe, UpperCasePipe, TitleCasePipe, KeyValuePipe, KeyValue} from './pipes/index';
```
Comment (AndrewKushnir) [inline packages/common/src/common.ts:25]: We should also add `DatePipeConfig` symbol here, so that it makes it into the public API.

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for the update, I've noticed that the `DatePipeConfig` symbol is not in the public API and I left a comment about it.

Also, the CI is failing due to the following error:
```
FAIL: Commit 656cf7a8d301a82dda03e1ffb26ec93bb25479a4 uncompressed
main exceeded expected size by 500 bytes or >1% (expected: 124269, actual: 124779).
```
Could you please update the `goldens/size-tracking/integration-payloads.json` file with the actual value (mentioned in the error message)?

Before the final review, could you please also merge all commits into one?

Please let us know once it's completed and I can take a final look.

Thank you.

**Comment by matthiasweiss** [comment]
@AndrewKushnir I have two questions:

1) Should I merge ALL commits into one or should the deprecation changes have their own commit?
2) I don't really know what to change in `goldens/size-tracking/integration-payloads.json`, the part that I would assume is relevant looks as follows:

```typescript
 "cli-hello-world-ivy-i18n": {
    "uncompressed": {
      "runtime": 926,
      "main": 124860,
      "polyfills": 35246
    }
  }
```

The size for main here would be larger than the size of the build in the pipeline, so I don't really know why it fails and how this could be fixed.

**Comment by AndrewKushnir** [comment]
@matthiasweiss please see my comment below.

> Should I merge ALL commits into one or should the deprecation changes have their own commit?

Yes, let's merge all commits into one. The deprecation message can be added to the end of the commit message (in this case it'd be picked up by the tooling and added into the changelog).

> I don't really know what to change in goldens/size-tracking/integration-payloads.json, the part that I would assume is relevant looks as follows:

The location that you mentioned is correct. Here is the current value in the main branch (which is aligned with what CI has): https://github.com/angular/angular/blob/main/goldens/size-tracking/integration-payloads.json#L29. Try rebasing the PR on top of the most up-to-date main branch and replace `124269` with `124779` in that file and add it to the PR.

Please let me know if any additional information is needed, I'll be happy to help.

Thank you.

**Comment by matthiasweiss** [comment]
@AndrewKushnir thanks for the quick reply, the suggestions are now worked in and the commits have been merged into one 👍

File: packages/common/src/pipes/date_pipe.ts [hunk 18]
```diff
@@ -10,14 +10,51 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
  * If the value isn't provided, the `DatePipe` will use the end-user's local system timezone.
+ *
+ * @deprecated use DATE_PIPE_DEFAULT_OPTIONS token to configure DatePipe
  */
 export const DATE_PIPE_DEFAULT_TIMEZONE = new InjectionToken<string>('DATE_PIPE_DEFAULT_TIMEZONE');
 
+/**
+ * DI token that allows to provide default configuration for the `DatePipe` instances in an
+ * application. The value is an object which can include the following fields:
+* - `dateFormat`: configures the default date format. If the not provided, the `DatePipe`
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:27]:
```suggestion
* - `dateFormat`: configures the default date format. If not provided, the `DatePipe`
```

File: packages/common/src/pipes/date_pipe.ts [hunk 19]
```diff
@@ -10,14 +10,51 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
+ * will use the 'mediumDate' as a value.
+ * - `timezone`: configures the default timezone. If the value isn't provided, the `DatePipe` will
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:29]: (for consistency with the previous sentence)
```suggestion
 * - `timezone`: configures the default timezone. If not provided, the `DatePipe` will
```

File: packages/common/src/pipes/date_pipe.ts [hunk 20]
```diff
@@ -10,14 +10,51 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
+ * use the end-user's local system timezone.
+ *
+ * @see `DatePipeConfig`
+ *
+ * @usageNotes
+ *
+ * Various date pipe default values can be overwritten by providing this interface.
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:36]:
```suggestion
 * Various date pipe default values can be overwritten by providing this token with the
 * value that has this interface.
```

**Review by AndrewKushnir — COMMENTED** [review]
@matthiasweiss thanks for the changes and merging commits into one.

I've left a few minor suggestions regarding docs. I'd also like to propose updating a commit message, here is a draft version:

```
feat(common): add injection token for default DatePipe configuration

This commit introduces a new `DATE_PIPE_DEFAULT_OPTIONS` token,
which can be used to configure default DatePipe options, such as date
format and timezone.

DEPRECATED:

The `DATE_PIPE_DEFAULT_TIMEZONE` token is now deprecated in favor
of the `DATE_PIPE_DEFAULT_OPTIONS` token, which accepts an object
as a value and the timezone can be defined as a field (called `timezone`)
on that object.
```

**Comment by matthiasweiss** [comment]
@AndrewKushnir I've applied your suggestions and changed the commit message, thanks again!

File: packages/common/src/pipes/date_pipe.ts [hunk 21]
```diff
@@ -10,14 +10,52 @@ import {Inject, InjectionToken, LOCALE_ID, Optional, Pipe, PipeTransform} from '
...
+ * ]
+ * ```
+ *
+ * Override the default timezone by providing a value using the token:
+ * ```typescript
+ * providers: [
+ *   {provide: DATE_PIPE_DEFAULT_OPTIONS, useValue: {timezone: '-1200'}}
+ * ]
+ * ```
+
```
Comment (AndrewKushnir) [inline packages/common/src/pipes/date_pipe.ts:54]: (extra line)
```suggestion
```

**Review by AndrewKushnir — APPROVED** [review]
@matthiasweiss thanks again for all the changes 👍

I've just left a minor comment, but overall the change looks good.

The next steps for this change:
- could you please address that small comment?
- we'll need a couple more approvals from the `public-api` group (there might be additional feedback/comments)
- we'll run the necessary tests in Google's codebase and let you know the result

Thank you.

**Comment by matthiasweiss** [comment]
@AndrewKushnir should be ready now! Looking forward to the feedback

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-common

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: public-api, size-tracking

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit bdb5371033d8e9a110619861323e8383a32d5900.

**Comment by AndrewKushnir** [comment]
@matthiasweiss thanks again for all your help with this feature 👍 

The PR was merged and this feature will be available to all Angular users as a part of upcoming v15 release 🎉 

Thanks for contributing to Angular!

**Comment by AndrewKushnir** [comment]
@matthiasweiss thanks again for contributing this feature to Angular 👍 

We want to mention this feature and yourself as an author in an upcoming blog post (about v15 release). Please let us know if that's ok or you'd prefer not to be mentioned in the blog post.

// cc @mgechev

**Comment by matthiasweiss** [comment]
@AndrewKushnir @mgechev Unfortunately I did not see your comment in time 😢 (a co-worker just showed me now)

If there is a chance that the blog post could be updated, then I'd love to be mentioned!

**Pull request closed without merging**